### ISSUE TITLE
Run::getProcessParameterSet() functionality disabled

### ISSUE THREAD

**Issue by PetrilloAtWork** [art-framework-suite/art#121]
I am trying to access the past job FHiCL configuration as early as possible.
The second best option for that is at the beginning of a run (the best one would have been at file opening time).

`art::DataViewImpl` provides a `getProcessParameterSet()`, and `art::Run` is `using` it.
Nevertheless I am failing to get a result from it: it consistently returns `false`.

I have stolen the code of its implementation, and it appears to work correctly on my input files... so I suppose it should work.
I would like to know if there is a reason why this functionality is instead disabled for run and subrun principals (that's my interpretation of the line `if (branchType_ != InEvent) { return false; }`), and if there is none, I would like it to be enabled.

**Comment by PetrilloAtWork** [comment]
For the record, the same functionality can be indirectly obtained already, for example:
```cpp
  for (art::ProcessConfiguration const& procConfig: run.processHistory()) {
    
    fhicl::ParameterSet config;
    if (!fhicl::ParameterSetRegistry::get(procConfig.parameterSetID(), config)) {
      throw cet::exception{ "ConfigExtractor" }
        << "Configuration of process '" << procConfig.processName()
        << "' can't be found!\n";
    }
    mf::LogInfo{ "ConfigExtractor" }
      << "Configuration of '" << procConfig.processName() << "' (release: "
      << procConfig.releaseVersion() << ", ID=" << procConfig.id() << "):"
      << '\n' << config.to_indented_string()
      ;
    
  } // for
```
prints all process configuration from a `art::Run`.

**Comment by knoepfel** [comment]
I agree that it is an unnecessary restriction to disable the function for (sub)runs.  Can I ask why you need to access the process parameter set?

**Comment by PetrilloAtWork** [comment]
It is currently the only was we have to discover the configuration of DAQ of ICARUS runs.
While this might be changed by having the DAQ write the configuration information _also_ as a run data product, that has not happened so far, so we will still need the access to the configuration.
However, note that even the data product solution is not particularly elegant, since it would likely consist in dumping the FHiCL configuration into a string data product. In my opinion, a real configuration data product is not feasible because the configuration is changing too rapidly for the data product class to keep up.

**Comment by knoepfel** [comment]
This will be included in art 3.11.
